arangod/IResearch/GeoAnalyzer.cpp

```diff
@@ -226,31 +245,42 @@ Result fromVelocyPack(velocypack::Slice object,
 
 Result fromVelocyPack(velocypack::Slice object,
                       GeoVPackAnalyzer::Options& options) {
-  auto r = fromVelocyPackBase(object, options);
-  if (!r.ok()) {
+  if (auto r = fromVelocyPackBase(object, options); !r.ok()) {
     return r;
   }
   return getBool(object, kLegacyParam, options.legacy);
 }
 
 Result fromVelocyPack(velocypack::Slice object,
                       GeoS2Analyzer::Options& options) {
-  auto r = fromVelocyPackBase(object, options);
-  if (r.ok()) {
+  if (auto r = fromVelocyPackBase(object, options); !r.ok()) {
     return r;
   }
-  auto value = object.get(kCompressionParam);
-  if (!value.isNone()) {
-    uint8_t v = 0;
-    if (!value.isNumber<uint8_t>() ||
-        (v = value.getNumber<uint8_t>()) <
-            static_cast<uint8_t>(s2coding::CodingHint::COMPACT)) {
-      return {TRI_ERROR_BAD_PARAMETER,
-              absl::StrCat("'", kCompressionParam, "' should be 0 or 1.")};
-    }
-    options.hint = static_cast<s2coding::CodingHint>(v);
+  auto value = object.get(kFormatParam);
+  if (value.isNone()) {
+    return {};
   }
-  return r;
+  auto error = [&] {
+    return Result{TRI_ERROR_BAD_PARAMETER,
+                  absl::StrCat("'", kFormatParam,
+                               "' should be one of strings: 's2Point', "
+                               "'latLngDouble', 'latLngInt'.")};
+  };
+  if (!value.isString()) {
+    return error();
+  }
+  auto it = kStr2Options.find(value.stringView());
+  if (it == kStr2Options.end()) {
+    return error();
+  }
+  options.coding = it->second;
+  LOG_TOPIC_IF("39b12", INFO, TOPIC,
```

Comment:
That message would be printed each time analyzer is created (and we do create analyzer many times during operations). Logs would be polluted. We could place this note in the docs actually.

Reply:
Ok